Use systemd-resolve to check if resolved is running

resolved may be included in nsswitch.conf even though systemd-resolved
is not running. E.g. Arch Linux includes resolved in nsswitch.conf by
default, but systemd-resolved is not enabled by default, which causes
this script to fail updating dns.

This patch uses the systemd-resolve command, which is included with
systemd, to check if systemd-resolved is actually running.
---
 vpnc-script | 3 +--
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/vpnc-script b/vpnc-script
index 6a55546..b00dbb9 100755
--- a/vpnc-script
+++ b/vpnc-script
@@ -117,19 +117,12 @@
 	ifconfig_syntax_ptpv6=""
 fi
 
-grep ^hosts /etc/nsswitch.conf 2>/dev/null|grep resolve >/dev/null 2>&1
-if [ $? = 0 ];then
-	RESOLVEDENABLED=1
-else
-	RESOLVEDENABLED=0
-fi
-
 if [ -r /etc/openwrt_release ] && [ -n "$OPENWRT_INTERFACE" ]; then
         . /etc/functions.sh
 	include /lib/network
 	MODIFYRESOLVCONF=modify_resolvconf_openwrt
 	RESTORERESOLVCONF=restore_resolvconf_openwrt
-elif [ -x /usr/bin/busctl ] && [ ${RESOLVEDENABLED} = 1 ]; then  # For systemd-resolved (version 229 and above)
+elif [ -x /usr/bin/busctl ] && grep -qs '^hosts.*resolve' /etc/nsswitch.conf && systemd-resolve --status >/dev/null 2>&1; then # For systemd-resolved (version 229 and above)
 	MODIFYRESOLVCONF=modify_resolved_manager
 	RESTORERESOLVCONF=restore_resolved_manager
 elif [ -x /sbin/resolvconf ]; then # Optional tool on Debian, Ubuntu, Gentoo and FreeBSD
